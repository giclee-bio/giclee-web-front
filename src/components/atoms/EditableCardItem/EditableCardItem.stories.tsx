import { Meta, StoryObj } from '@storybook/react'

import EditableCardItem from './EditableCardItem'

const meta: Meta<typeof EditableCardItem> = {}

export default meta
type Story = StoryObj<typeof EditableCardItem>

export const Index: Story = {
  args: {},
}
