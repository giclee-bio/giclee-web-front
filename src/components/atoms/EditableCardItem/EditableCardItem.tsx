import * as RadixForm from '@radix-ui/react-form'
import { DragHandleHorizontalIcon } from '@radix-ui/react-icons'
import Image from 'next/image'

import type { PlatformItems } from '@/components/atoms/PlatformItem'

interface Props extends PlatformItems {
  handleChangeText: (e: React.ChangeEvent<HTMLInputElement>, caption: string) => void
  handleClickDeleteItem: (caption: string) => void
}

const EditableCardItem: React.FC<Props> = ({
  alt,
  caption,
  platformAccountName,
  src,
  handleChangeText,
  handleClickDeleteItem,
}) => {
  return (
    <>
      <li className='mt-6 flex list-none items-center justify-between rounded-lg bg-stone-800 p-4 pl-8 first:mt-0'>
        <div className=''>
          <RadixForm.Field className='mt-4 first:mt-0' name={caption}>
            <div className='flex justify-between text-sm'>
              <RadixForm.Label className='font-bold'>{caption}</RadixForm.Label>
              <RadixForm.Message className='text-ruby-400' match='valueMissing'>
                必須項目です
              </RadixForm.Message>
            </div>
            <div className='mt-2 flex gap-2'>
              <Image alt={alt} className='rounded' height='32' src={src} width='32' />
              <RadixForm.Control asChild className='bg-stone-50'>
                <input
                  className='h-8 w-72 rounded px-3 text-black'
                  onChange={(e) => handleChangeText(e, caption)}
                  required
                  type='text'
                  value={platformAccountName}
                />
              </RadixForm.Control>
            </div>
          </RadixForm.Field>
          <div className='mt-2 flex justify-end'>
            <button
              className='text-sm text-ruby-400'
              onClick={() => handleClickDeleteItem(caption)}
            >
              Delete
            </button>
          </div>
        </div>
        <DragHandleHorizontalIcon className='text-stone-400' height='24' width='24' />
      </li>
    </>
  )
}

export default EditableCardItem
