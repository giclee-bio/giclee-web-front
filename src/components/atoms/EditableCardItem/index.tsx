import EditableCardItem from './EditableCardItem'

export default EditableCardItem
