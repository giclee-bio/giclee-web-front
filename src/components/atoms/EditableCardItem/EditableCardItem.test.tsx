import { render } from '@testing-library/react'

import EditableCardItem from './EditableCardItem'

import type { PlatformItems } from '@/components/atoms/PlatformItem'

const item: PlatformItems = {
  alt: 'nacal',
  caption: 'nacal',
  platformAccountName: 'nacal',
  src: '/nacal.png',
}

describe('EditableCardItem', () => {
  it('snapshot', async () => {
    const { asFragment } = render(
      <EditableCardItem {...item} handleChangeText={() => jest.mock} />,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
