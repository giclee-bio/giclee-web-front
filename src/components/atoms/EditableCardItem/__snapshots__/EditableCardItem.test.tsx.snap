// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableCardItem snapshot 1`] = `
<DocumentFragment>
  <form>
    <li
      class="mt-6 flex list-none items-center justify-between rounded-lg bg-stone-800 p-4 pl-8 first:mt-0"
    >
      <div
        class=""
      >
        <div
          class="mt-4 first:mt-0"
        >
          <div
            class="flex justify-between text-sm"
          >
            <label
              class="font-bold"
              for="radix-:r0:"
            >
              nacal
            </label>
          </div>
          <div
            class="mt-2 flex gap-2"
          >
            <img
              alt="nacal"
              class="rounded"
              data-nimg="1"
              decoding="async"
              height="32"
              loading="lazy"
              src="/_next/image?url=%2Fnacal.png&w=64&q=75"
              srcset="/_next/image?url=%2Fnacal.png&w=32&q=75 1x, /_next/image?url=%2Fnacal.png&w=64&q=75 2x"
              style="color: transparent;"
              width="32"
            />
            <input
              class="bg-stone-50 h-8 w-72 rounded px-3 text-black"
              id="radix-:r0:"
              name="nacal"
              required=""
              title=""
              type="text"
              value="nacal"
            />
          </div>
        </div>
        <div
          class="mt-2 flex justify-end"
        >
          <button
            class="text-sm text-ruby-400"
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
      <svg
        class="text-stone-400"
        fill="none"
        height="24"
        viewBox="0 0 15 15"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M2.49998 4.09998C2.27906 4.09998 2.09998 4.27906 2.09998 4.49998C2.09998 4.72089 2.27906 4.89998 2.49998 4.89998H12.5C12.7209 4.89998 12.9 4.72089 12.9 4.49998C12.9 4.27906 12.7209 4.09998 12.5 4.09998H2.49998ZM2.49998 6.09998C2.27906 6.09998 2.09998 6.27906 2.09998 6.49998C2.09998 6.72089 2.27906 6.89998 2.49998 6.89998H12.5C12.7209 6.89998 12.9 6.72089 12.9 6.49998C12.9 6.27906 12.7209 6.09998 12.5 6.09998H2.49998ZM2.09998 8.49998C2.09998 8.27906 2.27906 8.09998 2.49998 8.09998H12.5C12.7209 8.09998 12.9 8.27906 12.9 8.49998C12.9 8.72089 12.7209 8.89998 12.5 8.89998H2.49998C2.27906 8.89998 2.09998 8.72089 2.09998 8.49998ZM2.49998 10.1C2.27906 10.1 2.09998 10.2791 2.09998 10.5C2.09998 10.7209 2.27906 10.9 2.49998 10.9H12.5C12.7209 10.9 12.9 10.7209 12.9 10.5C12.9 10.2791 12.7209 10.1 12.5 10.1H2.49998Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </li>
  </form>
</DocumentFragment>
`;
